toxqnl/tqnl.c

```diff
@@ -0,0 +1,119 @@
+#include "tqnl.h"
+
+#include "query.h"
+
+#include "../toxcore/DHT.h"
+
+#define SET_ERROR(param, x) {if(param) {*param = x;}}
+
+struct Tox_QNL *tox_qnl_new(Tox *tox, TOX_QNL_ERR_NEW *error)
+{
+    if (!tox) {
+        SET_ERROR(error, TOX_QNL_ERR_NEW_TOX_NULL);
+        return NULL;
+    }
+
+    Messenger *m = (Messenger *)tox;
+
+    if (!m->net || !m->dht) {
+        SET_ERROR(error, TOX_QNL_ERR_NEW_TOX_INVALID);
+        return NULL;
+    }
+
+    struct Tox_QNL *tqnl = (Tox_QNL *)calloc(1, sizeof(*tqnl));
```

Comment:
wait, it seems correct, but it is still confusing, better use `sizeof(Tox_QNL)`?